### PR TITLE
Refactor wheels 538

### PR DESCRIPTION
**Pull request by steamypotato**
# Assignee Section

## Description

Refactored wheel code. 

### Steps for Testing

Build the code and move the wheels.

closes #[538]

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [x] Local Test Performed Successfully
- [x] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

**Comment by RealWilliamWells** [comment]
Just built and tested the code from the latest commit from this PR and it worked perfectly for controlling the wheels. Great work gamer! Later on I will do some code review on my own, but you and Tim should also do it together this weekend.

File: robot/rover/Arm/Arm.ino [hunk 1]
```diff
@@ -188,10 +188,7 @@ void setup() {
   \todo There should be a check so that if the motor is moving away from the goal position
   or has been moving for a while without reaching the expected angle, it stops...
   like a timeout.
-  \todo What happens if a new command tells the motor to turn in opposite direction? Abrupt changes are bad.
-  \todo If the stepper is trying to turn but hasn't gotten anywhere, there should be
-  a check in the microcontroller that there's an issue (there can also be a check in the gui)
-  \todo Check to see if any global variables can be turned into static variables inside loop()
+  \todo What happens if a    variables inside loop()
```
Comment (VimFreeman) [inline robot/rover/Arm/Arm.ino:191]: Please revert this file as it's unrelated. We don't want to introduce potentially unwanted changes

File: robot/rover/MobilePlatform/MobilePlatform.ino [hunk 2]
```diff
@@ -25,175 +11,84 @@ const uint8_t RX_TEENSY_3_6_PIN = 0;
...
 }
-
 void WheelsCommandCenter::stopMotors() {
     Rover::stopMotors();
-    Rover::systemStatus.last_throttle = millis();
 }
 
-void WheelsCommandCenter::closeMotorsLoop() {
-    Rover::closeLoop();
+void WheelsCommandCenter::moveRover(const float & linear_y,const float & omega_z) {
```
Comment (VimFreeman) [inline robot/rover/MobilePlatform/MobilePlatform.ino:83]: Did you get this working in the end?

File: robot/rover/MobilePlatform/includes/commands/WheelsCommandCenter.h [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
-#define COMMAND_SET_ACCELERATION 7
-#define COMMAND_GET_ROVER_STATUS 8
-#define COMMAND_MOVE_ROVER 9
-#define COMMAND_MOVE_WHEEL 10
+#include "CommandCenter.h"
+
+#define COMMAND_STOP_MOTORS_EMERGENCY 0
+#define COMMAND_GET_ROVER_STATUS 2
+#define COMMAND_SEND_GPS 3
+#define COMMAND_MOVE_SERVO 7
```
Comment (VimFreeman) [inline robot/rover/MobilePlatform/includes/commands/WheelsCommandCenter.h:9]: Can you please change all the command IDs so that they are sequential? And reflect the changes in WheelCommands.py

File: robot/rover/MobilePlatform/src/commands/WheelsCommandCenter.cpp [hunk 4]
```diff
@@ -24,111 +13,87 @@ float bytes_to_float(const uint8_t* rawPointer)
...
+            moveRover(linear,angular);
+
+            break;
+        }
+        case COMMAND_MOVE_WHEEL:{
+            uint8_t wheelNumber = (*rawArgs++);
+            uint8_t direction = (*rawArgs++);
+            uint8_t speed = (*rawArgs++);
+            moveWheel(wheelNumber,direction,speed);
+            break;
```
Comment (VimFreeman) [inline robot/rover/MobilePlatform/src/commands/WheelsCommandCenter.cpp:48]: Love this

File: robot/rover/PDS/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,64 @@
+file(MAKE_DIRECTORY build)
```
Comment (VimFreeman) [inline robot/rover/PDS/CMakeLists.txt:1]: Does this file need to be in this PR?

File: robot/rover/PDS/PDS2/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,64 @@
+file(MAKE_DIRECTORY build)
```
Comment (VimFreeman) [inline robot/rover/PDS/PDS2/CMakeLists.txt:1]: Also this file

File: robot/rover/comms_demo/arduino_demos/pong_bing/pong_bing.ino [hunk 7]
```diff
@@ -1,26 +1,36 @@
-#include <Arduino.h>
-uint32_t currentTime;
+#include "Arduino.h"
```
Comment (VimFreeman) [inline robot/rover/comms_demo/arduino_demos/pong_bing/pong_bing.ino:1]: what is this

File: robot/rover/comms_demo/comms_demo.ino [hunk 8]
```diff
@@ -1,29 +1,28 @@
-// USB : Debug, UART : Production
-#define USB
+#include "Arduino.h"
```
Comment (VimFreeman) [inline robot/rover/comms_demo/comms_demo.ino:1]: also this :')

**Review by VimFreeman — CHANGES_REQUESTED** [review]
Pretty good work. Much cleaner than before. 
I made a few minor comments to resolve. We'll do some thorough testing as well to make sure that there are no weird edge-cases.

File: robot/rover/comms_demo/arduino_demos/pong_bing/pong_bing.ino (hunk 7 above)
Comment (steamypotato) [reply robot/rover/comms_demo/arduino_demos/pong_bing/pong_bing.ino:1]: random leftover test file that someone created a long ass time ago

File: robot/rover/comms_demo/comms_demo.ino (hunk 8 above)
Comment (steamypotato) [reply robot/rover/comms_demo/comms_demo.ino:1]: random leftover test file that someone created a long ass time ago

File: robot/rover/MobilePlatform/MobilePlatform.ino (hunk 2 above)
Comment (steamypotato) [reply robot/rover/MobilePlatform/MobilePlatform.ino:83]: need to test it out a bit again(this is for twist commands)

File: robot/rover/PDS/CMakeLists.txt (hunk 5 above)
Comment (steamypotato) [reply robot/rover/PDS/CMakeLists.txt:1]: really old stuff. shall terminate.

File: robot/rover/PDS/PDS2/CMakeLists.txt (hunk 6 above)
Comment (steamypotato) [reply robot/rover/PDS/PDS2/CMakeLists.txt:1]: really old stuff. shall terminate.

File: robot/rover/Arm/Arm.ino (hunk 1 above)
Comment (steamypotato) [reply robot/rover/Arm/Arm.ino:191]: i will just copy and paste the ino file from master

File: robot/rover/MobilePlatform/includes/commands/WheelsCommandCenter.h (hunk 3 above)
Comment (steamypotato) [reply robot/rover/MobilePlatform/includes/commands/WheelsCommandCenter.h:9]: I will remove all the un-used commands until they actually work (like the velocity stuff until encoders work again)

File: robot/rover/MobilePlatform/MobilePlatform.ino (hunk 2 above)
Comment (steamypotato) [reply robot/rover/MobilePlatform/MobilePlatform.ino:83]: like i think the direction needs to be reversed and the sensitivity needs to be adjusted iirc

**Review by VimFreeman — COMMENTED** [review]
Can you confirm that everything is ready to be merged? Should I do one final look over?

**Comment by steamypotato** [comment]
@VimFreeman Yeah I think its good
